content/news/2022-07-29-bevy-0.8/index.md

```diff
@@ -0,0 +1,1131 @@
+
+## Automatic Mesh Tangent Generation
+
+<div class="release-feature-authors">authors: Rob Swain (@superdump), @DJMcNab</div>
+
+Vertex tangents are used in tandem with normal maps to give meshes more detailed normals when rendering them. Some imported meshes have normal maps, but don't have vertex tangents calculated. Bevy can now automatically generate vertex tangents for [`Meshes`][`Mesh`] that are missing them using the defacto industry-standard MikkTSpace library / algorithm (Godot, Unity, Unreal, and Blender all use this).
+
+We have started maintaining [our own fork](https://github.com/bevyengine/bevy/tree/v0.8.0/crates/bevy_mikktspace) of the [gltf-rs/mikktspace crate](https://github.com/gltf-rs/mikktspace) so we can:
+1. Update dependencies at the speed required for Bevy
+2. [Start reining in the unsafe code](https://github.com/bevyengine/bevy/pull/4932) (it currently uses unsafe rust code auto-generated from the original mikktspace.h written in C).
+
+## Default to Linear Texture Filtering
+
+<div class="release-feature-authors">authors: @aevyrie, @cart</div>
+
+Images in Bevy now use linear texture filtering by default, which is more in line with the rest of the gamedev ecosystem (Unity and Godot both default to filtered textures). 
+
+This means that textures that require unfiltered pixels (such as "pixel art" sprites) must override this default, either per-image:
+
+```rust
+image.sampler_descriptor = ImageSampler::nearest();
+```
+
+or globally using the new [`ImageSettings`] resource:
+
+```rust
+app.insert_resource(ImageSettings::default_nearest())
+```
+
+With that, we get crisp pixel art:
+
+![sprite](sprite.png)
+
+[`ImageSettings`]: https://docs.rs/bevy/0.8.0/bevy/render/texture/struct.ImageSettings.html
+
+## New GlobalTransform Matrix Representation
+
+<div class="release-feature-authors">authors: @HackerFoo</div>
+
+The internal representation of the [`GlobalTransform`] component (representing the "world space" transform for an entity) has been changed from a "similarity" (translation [`Vec3`] / rotation [`Quat`] / scale [`Vec3`]) to an "affine 3d transform" ([`Mat3A`] and a [`Vec3`] translation).
+
+Notably, this allows for shear to be represented. Shear is a controversial topic. Engine and physics programmers tend to hate it. Artists tend to love it. Given that most artist tools and game engines support shear in their equivalent types, we believe it is important to provide this as an option. 
+
+[`Mat3A`]: https://docs.rs/bevy/0.8.0/bevy/math/struct.Mat3A.html
+[`Vec3`]: https://docs.rs/bevy/0.8.0/bevy/math/struct.Vec3.html
+[`Vec3A`]: https://docs.rs/bevy/0.8.0/bevy/math/struct.Vec3A.html
+[`Quat`]: https://docs.rs/bevy/0.8.0/bevy/math/struct.Quat.html
+
+## AABB Recalculation
+
+<div class="release-feature-authors">authors: @mlodato517</div>
+
+When a [`Mesh`] asset is changed, the AABB (axis aligned bounding box) is now recalculated, ensuring frustum culling calculations are still correct.
```

Comment:
I believe this had a serious impact on animated meshes, which should be called out.

Reply:
Not really. Only if you hand-animate the vertices. For meshes animated through skinning, the animation of the vertices happens in the vertex shader. We talked about having to calculate conservative Aabbs that contain the maximal extents of the animation, but that hasn’t been implemented. So it’s rather a known issue that animated meshes may be visible or culled when they should be the opposite.